src/server/db/actions/TagAction.ts

```diff
@@ -1,36 +1,62 @@
 import TagModel from "../models/TagModel";
 import connectMongoDB from "../mongodb";
-import { ObjectId } from "mongodb";
-import { ITag } from "../models/TagModel";
 import GameModel from "../models/GameModel";
-import { GenericUserErrorException } from "@/utils/exceptions";
-
-export async function createTag(data: ITag) {
-  connectMongoDB();
-
-  const tag = new TagModel(data);
+import { TagNotFoundException } from "@/utils/exceptions/tag";
+import { CreateTagInput } from "@/pages/api/tags";
 
+export async function createTag(data: CreateTagInput) {
+  await connectMongoDB();
+  const session = await TagModel.startSession();
+  session.startTransaction();
```

Comment:
when should we use this vs not use this?

Reply:
Transactions are helpful for when you want to mutate things atomically - for example, like when you need all the database operations to succeed in order to mutate multiple documents.

In this case, we need createTag to create a tag, get its id, then modify the games that it applies to by adding the id to their `tags` array. Say for some reason, the game modification part fails. If we didn't do this atomically, then we would still be left with a newly created tag even though the downstream events failed. Transactions prevent this sort of side-effect behavior from happening by only `commit`ing the transaction when everything is successful. This is why you see `abortTransaction` in the catch block, so that none of the database mutations we do inside of the transaction are persisted. Hopefully that makes sense!